api/query/query.go

```diff
@@ -142,6 +144,11 @@ func getMemcacheKey(testRun shared.TestRun) string {
 }
 
 func isRequestCacheable(r *http.Request) bool {
+	q := r.URL.Query()
+	if _, showMetadata := q[shared.MetadataFlag]; showMetadata {
```

Comment:
I would probably still cache the request, even if it's showing metadata; we don't cache forever or anything.
In either case, this would fall under the should-cache, not can-cache (they're separate).